phpunit/blocks/render-block-navigation-submenu-test.php

```diff
@@ -0,0 +1,213 @@
+<?php
+/**
+ * Tests server side rendering of core/navigation-submenu
+ *
+ * @package    Gutenberg
+ * @subpackage block-library
+ */
+
+/**
+ * Tests for various cases in Navigation Submenu rendering
+ */
+class Render_Block_Navigation_Submenu_Test extends WP_UnitTestCase {
```

Comment:
@scruffian What do you think of the test coverage?

Reply:
Should we use the navigation block itself in the tests rather than the navigation submenu block? I think I prefer using the submenu block as you have done, as it feels more inline with the future direction of the block.

The coverage looks good, but I have two test failures:
```
1) Render_Block_Navigation_Submenu_Test::test_should_apply_custom_colors_inherited_from_parent_block_via_context
2) Render_Block_Navigation_Submenu_Test::test_should_apply_mix_of_preset_and_custom_colors_inherited_from_parent_block_via_context
```